client/src/partials/exchange/modal.js

```diff
@@ -3,19 +3,34 @@ angular.module('bhima.controllers')
 
 ExchangeModalController.$inject = [
   'ExchangeRateService', 'CurrencyService',
-  'SessionService', '$uibModalInstance', 'data'
+  'SessionService', '$uibModalInstance', 'data', 'exchange'
 ];
 
-function ExchangeModalController(Rates, Currencies, Session, $uibModalInstance, data) {
+function ExchangeModalController(Rates, Currencies, Session, $uibModalInstance, data, exchange) {
   var vm = this;
 
   // bind variables
   vm.data = data;
+
   vm.submit = submit;
   vm.cancel = cancel;
   vm.enterpriseCurrencyId = Session.enterprise.currency_id;
   vm.enterpriseCurrency = format(Session.enterprise.currency_id);
   vm.selectedCurrency = format(data.currency_id);
+  vm.today = new Date();
+
+  vm.hasDailyRate = Rates.hasDailyRate(vm.data.currency_id,vm.data.date);
```

Comment:
This determines whether to require the insertion in case there is no exchange rate or the updated otherwise